Dockerfile

```diff
@@ -46,13 +46,17 @@ ENV STATE_DB_IMPL="geth"
 ENV VM_IMPL="geth"
 ENV LD_LIBRARY_PATH=./
 
-EXPOSE 5050
-EXPOSE 6060
-EXPOSE 18545
-EXPOSE 18546
+EXPOSE 5050 6060 18545 18546
 
 COPY genesis/example-genesis.json ./genesis.json
 COPY scripts/run_sonic_privatenet.sh ./run_sonic.sh
 COPY scripts/set_genesis.sh ./set_genesis.sh
 
+# Add https://github.com/krallin/tini
+ENV TINI_VERSION=v0.19.0
+ADD https://github.com/krallin/tini/releases/download/${TINI_VERSION}/tini /tini
+RUN chmod +x /tini
+
+# -g forwards SIGINT to child processes and terminates client gracefully
+ENTRYPOINT ["/tini", "-g", "--"]
```

Comment:
what is this good for? I understand that it wraps the process to another container? I fear that it can make debugging more complicated, and only hide other problems. If this is suppose to fix graceful shutdown, I think we should have a look why the client is not carefully shoot.

Reply:
For process in docker container to capture signals, it needs to be run with PID 1. 
The common way to do this is to do `docker run --init`, equivalently this [init config in HostConfig](https://github.com/moby/moby/blob/32e2c05bb6c9d039275de7f07c3255059cfd88c4/api/types/container/hostconfig.go#L480).
The HostConfig didn't work so I tried to build that in.

Reply:
Change removed

Reply:
Is this solving issue described here? https://docs.docker.com/reference/cli/docker/container/kill/

If yes, it should be documented, because it is not that obvious

Reply:
It does not.

Before the change:
`docker container kill --signal=SIGINT <target-container>` was ignored
`docker container kill --signal=SIGHUP <target-container>` would result in dirty DB
`docker container kill --signal=SIGKILL <target-container>` would result in dirty DB

Reply:
I understand the need for documentation - the changes are no longer introduced.
The next time it's pull-requested, it will be with a documentation.